src/sdk/pynni/nni/msg_dispatcher.py

```diff
@@ -111,11 +111,20 @@ def handle_add_customized_trial(self, data):
 
     def handle_report_metric_data(self, data):
         if data['type'] == 'FINAL':
+            value = None
             id_ = data['parameter_id']
+            
+            if isinstance(data['value'], float) or isinstance(data['value'], int):
+                value = data['value']
+            elif isinstance(data['value'], dict) and 'default' in data['value']:
+                value = data['value']['default']
+            else:
+                raise RuntimeError('The final result from Trial is dict and has no default key in Tuner.')  
```

Comment:
`RuntimeError('Incorrect final result: the final result should be float/int, or a dict which has key named "default"')`